src/main/java/net/smartcosmos/edge/things/resource/GetThingResource.java

```diff
@@ -71,4 +74,17 @@ public GetThingResource(GetThingEdgeService getThingService) {
 
         return getThingService.getByType(type, fields, page, size, sortOrder, sortBy, user);
     }
+
+    @ConditionalOnProperty(prefix = ENDPOINT_ENABLEMENT_THINGS_READ_TYPE, name = ENDPOINT_ENABLEMENT_PROPERTY_ENABLED, matchIfMissing = true)
+    @RequestMapping(value = ENDPOINT_TYPE_URNS,
+                    method = RequestMethod.POST,
+                    produces = APPLICATION_JSON_UTF8_VALUE,
+                    consumes = APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<?> findByTypeAndUrns( // @formatter:off
+        @PathVariable(TYPE) String type,
+        @RequestBody @Valid Map<String, Set<String>> urns,
+        SmartCosmosUser user) { // @formatter:on
+
+        return getThingService.getByTypeAndUrns(type, urns, user);
```

Comment:
We need to escape/encode these values to ensure we do not encounter injection errors.  That should probably be done right here since this is the first location in the code we encounter the values and almost any where later might be too late.